### PR TITLE
Develop #6

### PR DESCRIPTION
**Pull request by rpitingolo**
@NeighborhoodInfoDC/review review request, please. This program creates HMDA base files for 2008-2017. One outstanding issue is what to do with missing tract IDs. By my calculation it affects about a quarter of a percent (in DC) each year.

### CODE REVIEW

**Comment by ptatian** [comment]
@rpitingolo Thanks. Reviewing this now.

File: Prog/Loans 2008-2017.sas [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+		   edit $edit.
+		   hoepa $hoepa.
+		   lien $lien.
+		   loantype loantyp.
+		   occupanc $occupan.
+		   prch_typ $prchtyp.
+		   preapp $preapp.
+		   purpose $purpose.
+		   type $proptyp.
+		   ucounty $cnty99f.
```
Comment (ptatian) [inline Prog/Loans 2008-2017.sas]: Should change _ucounty_ format to _$cnty15f._

File: Prog/Loans 2008-2017.sas [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+	seq = sequence_number;
+	type = put(property_type,1.);
+	county = put(county_code,z3.);
+	year = put(as_of_year,4.);
+	state = put(state_code,2.);
+	tract = compress(census_tract_number,".");
+
+	/* Combined ucounty and geo2000/geo2010 */
+	ucounty = state || county;
+
```
Comment (ptatian) [inline Prog/Loans 2008-2017.sas]: We only want to keep loans made in the metro area. You can use this code to select those records.

`if put( ucounty, $ctym15f. ) ^= " ";`

File: Prog/Loans 2008-2017.sas [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+	occupanc = put(owner_occupancy,1.);
+	prch_typ = put(purchaser_type,1.);
+	preapp = put(preapproval,1.);
+	purpose = put(loan_purpose,1.);
+	race2 = put(applicant_race_2,1.);
+	race3 = put(applicant_race_3,1.);
+	race4 = put(applicant_race_4,1.);
+	race5 = put(applicant_race_5,1.);
+	rtspread = rate_spread;
+	seq = sequence_number;
```
Comment (ptatian) [inline Prog/Loans 2008-2017.sas:282]: The value of _seq_ is entirely missing in 2017.

**Review by ptatian — CHANGES_REQUESTED** [review]
Thanks, @rpitingolo. I made a few comments. Let me know if you have any questions.

**Comment by ptatian** [comment]
Sorry, @rpitingolo. It occurred to me later that you don’t need to filter out the records without tracts from the **loan** file. You could just do this when you create the summary files.

**Comment by rpitingolo** [comment]
@ptatian what do you think about creating a flag for missing or invalid tract in the loan file and keeping everything? It could make it easier to filter out later? Or might be unnecessary?

**Comment by ptatian** [comment]
@rpitingolo Good idea. Thanks.

File: Prog/Loans 2008-2017.sas (hunk 2 above)
Comment (rpitingolo) [reply Prog/Loans 2008-2017.sas]: Thanks. I've added this.

File: Prog/Loans 2008-2017.sas (hunk 3 above)
Comment (rpitingolo) [reply Prog/Loans 2008-2017.sas:282]: I've done some digging but am not 100% clear on why the seq variable is missing sometimes (and all the time in 2017). Regardless, it seems like this variable is not used in the summary program, so if it's OK with you I'd like to finalize this and move on to updating the summary files.

**Pull request merged**